### PR TITLE
automated: linux: add docker-corrupt OTA test

### PR DESCRIPTION
**Pull request by brenwoo**
Verify successful rollback after OTA upgrade containing corrupt docker app.

Signed-off-by: Brendan Wood <brendan.wood@foundries.io>

### CODE REVIEW

File: automated/linux/ota-rollback/verify-rollback.sh [hunk 1]
```diff
@@ -85,6 +86,25 @@ else
...
+if [ "${TYPE}" = "app" ]; then
+    echo "Checking for broken app"
+    runtime="5 minute"
+    endtime=$(date -ud "$runtime" +%s)
+
+    while [ "$(date -u +%s)" -le "$endtime" ] && [ ! "${FOUND}" -eq 1 ]; do
+        journalctl --no-pager -u aktualizr-lite
+        echo "Sleep for 1 min"
+        sleep 60
+        if (journalctl --no-pager -u aktualizr-lite | grep "Finalization has failed; reason: apps start failure"); then
```
Comment (mwasilew) [inline automated/linux/ota-rollback/verify-rollback.sh:99]: maybe there is a way to determine the state by MESSAGE and RESULT variables from aklite?

**Review by mwasilew — CHANGES_REQUESTED** [review]

File: automated/linux/ota-rollback/verify-rollback.sh [hunk 2]
```diff
@@ -73,7 +74,7 @@ if [ -f /usr/lib/firmware/version.txt ]; then
     # shellcheck disable=SC2154
     ref_bootfirmware_version_after_rollback="${bootfirmware_version}"
     if [ "${TYPE}" = "uboot" ]; then
-        ref_bootfirmware_version_after_rollback=0
+    	ref_bootfirmware_version_after_rollback=0
```
Comment (mwasilew) [inline automated/linux/ota-rollback/verify-rollback.sh:77]: tabs vs. spaces

File: automated/linux/ota-rollback/verify-rollback.sh [hunk 3]
```diff
@@ -85,6 +86,26 @@ else
...
+            report_fail "app-rollback"
+            break
+        elif (journalctl --no-pager -u aktualizr-lite | grep "Finalization has failed; reason: apps start failure"); then
+            echo "broken app found"
+            FOUND=1
+            report_pass "app-rollback"
+        fi
+    done
+else
+	report_skip "app-rollback"
```
Comment (mwasilew) [inline automated/linux/ota-rollback/verify-rollback.sh:107]: too long indentation

**Review by mwasilew — CHANGES_REQUESTED** [review]
Looks very good. Just 2 minor edits and we're good to go.

**Comment by mwasilew** [comment]
LGTM

**Comment by roxell** [comment]
I'm ok with this too.

@brenwoo would it be cleaner to do http://ix.io/4iKg ? maybe you can do that in a followup patch?

**Comment by nareshkamboju** [comment]
@brenwoo  Thanks for your contribution.
@mwasilew and @roxell Thanks for the review.

**Pull request merged**